src/strings.cpp

```diff
@@ -806,18 +828,18 @@ static std::vector<const char *> _game_script_raw_strings;
 
 /**
  * Parse most format codes within a string and write the result to a buffer.
- * @param buff    The buffer to write the final string to.
+ * @param builder The string builder to write the final string to.
  * @param str_arg The original string with format codes.
  * @param args    Pointer to extra arguments used by various string codes.
- * @param last    Pointer to just past the end of the buff array.
  * @param dry_run True when the argt array is not yet initialized.
  */
-static char *FormatString(char *buff, const char *str_arg, StringParameters *args, const char *last, uint case_index, bool game_script, bool dry_run)
+static void FormatString(StringBuilder &builder, const char *str_arg, StringParameters *args, uint case_index, bool game_script, bool dry_run)
 {
 	uint orig_offset = args->offset;
 
 	/* When there is no array with types there is no need to do a dry run. */
 	if (args->HasTypeInformation() && !dry_run) {
+		StringBuilder dry_run_builder = builder;
```

Comment:
Codewise this feels off. The line above says `if not dry-run`, after which you make a `dry-run` variable. But also: why? The old code didn't do it?

As you can see, I don't actually understand the code, so it might be easily explained :)

Reply:
Added some comments in the code what is happening/why it is happening, but it that doesn't make it show as outdated here.